docs/DeveloperGuide.md

```diff
@@ -4,26 +4,162 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
-
+## Design
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Good Class
+#### Description
+The goods class keeps track of the various inventory that will be input into the system through the commands class.
+The diagram below shows the model component of the orderline class.
+![Good Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/Good.puml)
+
+The Goods class allows for the creation of orderline objects which has the following attributes: id, name, quantity and 
+description. Each attribute can be obtained using public get methods, and the attribute quantity can be set using the public set method.
+
+
+## Implementation
+### Regex Class
+#### Description
+The Regex Class is created to simplify the use of regular expressions in Java.
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/Regex.puml)
+The Class Diagram above shows the different variables and methods associated with the regex class.
+This class takes in the regular expression as a string and the string to be matched in the class contructor, and stores the results
+in the private variable `groupValues` which is a Hash Map where the key would be the capture group name and the value as the matched results.
```

Comment:
Consider using `HashMap` in a code block instead of Hash Map to make it clear that it is a class.
![image](https://user-images.githubusercontent.com/60597985/160962185-3c4c1944-4848-4eae-80ef-e141e06b7f7e.png)